Running your first OT on Ubuntu Linux

Ralumbi opened the pull request:
Created a book for setting up a complete webserver on linux

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,178 @@
+# Setting up your ubuntu linux server
+guiding to setup your very own ubuntu linux server with a webserver plus the ability to deploy your tfs server
```
Should be full sentences.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,178 @@
+# Setting up your ubuntu linux server
+guiding to setup your very own ubuntu linux server with a webserver plus the ability to deploy your tfs server
+
+> We are not responsible for the mistakes you make
```
I wouldn't even add that, it's obvious.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,178 @@
+> We are not responsible for the mistakes you make
+
+# Getting started
+Rent a ubuntu server 18.04 or higher from a trusted hosting company such as ovh.com or install for the sake of learning experience ubuntu server 18.04 or higher on another computer or if you are on windows use virtualbox to get your ubuntu server.
```
Avoid naming specific companies in the official docs (remove ovh.com)

DSpeichert replied:
Proper capitalization needed (`Ubuntu Server` not `ubuntu server`), also the operating system isn't really tied to the physical server you rent.

Ralumbi commented:
@DSpeichert I've changed the points you've pointed out, thanks for letting me know. Happy to learn things the more professional way. Also added the final part which I totally forgot to setup TFS.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,201 @@
+# Getting started
+Rent a Ubuntu Server 18.04 or higher from a trusted hosting company such or install for the sake of learning experience Ubuntu Server 18.04 or higher in VirtualBox.
+
+Once you're set we will start by accessing our server. If you're on windows you should get putty(https://www.putty.org/), if you're on macOS or Linux you can use the terminal which is standard available. With putty just follow the screen ( insert your domain/ip-address and click ok ) for macOS and linux users type:
```
I think it would be safer to recommend using Windows Subsystem for Linux as a native SSH client or simply installing SSH client via Windows features.

Ralumbi replied:
Here you have a point indeed! WSL is safer, but putty would be easier for new people. It would be your call what you would like to be better in there.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+```bash
+sudo apt install nginx php-fpm mariadb-server -y
+```
+The installation process is now triggered.When this is finished we will install phpmyadmin to handle our database server (which is mariadb)
````
Missing space, capitalization (`MariaDB`)

Ralumbi replied:
Removed the part as its not necessary to be there.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+```
+The output will be looking something similar as this (I am using ubuntu 19.10 here which uses php version 7.3 as default. If you're running on a server you most likely run on 18.04 lts or with the newest version 20.04 lts):
+
+![alt text](https://worldofcoding.net/github-img/otland-gitbook/phpversion.png "phpversion")
````
what is this website? It should ideally be just text, let's avoid images of text.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,201 @@
+
+![alt text](https://worldofcoding.net/github-img/otland-gitbook/phpversion.png "phpversion")
+
+Second we want to do is enabling php in our nginx server and setting the configuration just as we want. So we will be going to edit our nginx config. Now we need to install a command line text editor called "vim". After that we installed it we will directly open the configuration file.
```
`nano` is easier than `vim` but it's just a matter of preference.

Ralumbi replied:
I struggle with nano 🔢

Xaekai replied:
Unless you know your audience _already_ knows vi, nano is the safe bet. The audience that already knows vi probably wouldn't need this portion of the guide in the first place.

Though, if you are aiming for the neophytes and you know they are using modern systems, such as Ubuntu 20.04LTS, [micro](https://micro-editor.github.io/) should take precedence over nano.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+sudo apt install vim -y && sudo vim /etc/nginx/sites-available/default
+```
+
+Now sit straight up and read carefully on what we're going to do. I am going to give you a few options in the configuration. Underneath here you will find the default configuration file of nginx. ( I only took the part of the config file we are in need to use and left out all the comments which we do use in this tutorial )
````
> Now sit straight up and read carefully on what we're going to do.
While funny, this is meta-text and doesn't actually contribute anything. Let's avoid unnecessary prose.

Ralumbi replied:
Removed it.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,201 @@
+
+## Configuring php
+
+I already enabled php here by writing 'index.php' on line number '24'. I've uncommented the php location settings we need to use. To make it easy and understanding I made own comments for you guys to know which lines to comment and uncomment, depending on which Ubuntu Server version you are running ( and assuming you're using Ubuntu Server 18.04 lts ).
```
> Ubuntu Server 18.04 lts 
it's `LTS`

Ralumbi replied:
Changed

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+```bash
+sudo mariadb
+```
+![alt text](https://worldofcoding.net/github-img/otland-gitbook/phpversion.png "phpversion")
````
This again?

Ralumbi replied:
Removed it.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+`sudo vim /etc/nginx/sites-available/phpmyadmin`
+```bash
+server {
+    listen      2344;
````
Why this port?

Ralumbi replied:
Just took a random port, can be changed to what you desire.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+
+Git will be used to download the source code, and CMake will be used to generate the build files.
+```bash
+sudo apt-get install git cmake build-essential liblua5.2-dev libgmp3-dev libmysqlclient-dev libboost-system-dev libboost-iostreams-dev libboost-filesystem-dev libpugixml-dev libcrypto++-dev
````
This is outdated.

Ralumbi replied:
Changed.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,201 @@
+mkdir build && cd build
+cmake ..
+```
+## Build
````
Try to leave one empty line above every header.

Ralumbi replied:
Done 👍

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,201 @@
+# Getting started
+Rent a Ubuntu Server 18.04 or higher from a trusted hosting company such or install for the sake of learning experience Ubuntu Server 18.04 or higher in VirtualBox.
```
Technically, WSL could be used on Windows as well instead of VirtualBox.

Ralumbi replied:
Using WSL as server would cause many people with a lot of issues. I been through issues with it as well. Giving more options would give people a better feeling for themselves to pick the way they find to use the best. VirtualBox is given as example since its great for new people. When you've made a mistake you delete the VM and create a new one and start over. We are making guides for people which has no knowledge.

Xaekai replied:
VirtualBox also empowers them with advanced solutions, such as using USB passthrough to a separate Wifi dongle. This ensures Windows is not interfering, in any manner whatsoever, with the TFS servers internet connectivity. An important elimination of complexity in the typical diagnosing self-hosted login issues thread in the support forum. Though admittedly outside the scope of this particular guide.

Still, if both options are presented, VirtualBox should be preferred, because it presents the full native Linux environment. And if there is an advanced guide on the matter they follow later, it would definitely introduce them to VBox's seamless mode, which is a game changer for really experiencing the power of a Linux desktop environment without leaving their comfortable safety of Windows. Call it optimism for the best result.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+# Getting started
+Rent a Ubuntu Server 18.04 or higher from a trusted hosting company or install for the sake of learning experience Ubuntu Server 18.04 or higher in VirtualBox.
+
+Once you're set we will start by accessing our server. If you're on windows you should get putty(https://www.putty.org/), if you're on macOS or Linux you can use the terminal which is standard available. With putty just follow the screen ( insert your domain/ip-address and click ok ) for macOS and linux users type:
```
Lot's of typos:
* `PuTTY` not `putty`
* No spaces inside `(` and `)`
* `IP address` not `ip-address`
* `click "OK"` not `click ok`
* `Linux` not `linux`

ranisalt replied:
You can use a link to PuTTy instead

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+`username example => root
+domain/ipaddress example => yourdomain.com or 127.0.0.1`
+
+Now you will be prompted for your password (if in putty you will be prompted for the password as soon you click on Open) Insert your password and presh enter (you will not see what you type, so make sure you type your password in the right order). Now we are in our server :)
```
PuTTY misspelled
Missing full stop at the end of sentence.
`presh` -> `press`

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+Now you will be prompted for your password (if in putty you will be prompted for the password as soon you click on Open) Insert your password and presh enter (you will not see what you type, so make sure you type your password in the right order). Now we are in our server :)
+
+## Update and Upgrade the system
+The first thing we will do is updating our Ubuntu Server with the following command
```
`command:` (colon missing)

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+Now you will be prompted for your password (if in putty you will be prompted for the password as soon you click on Open) Insert your password and presh enter (you will not see what you type, so make sure you type your password in the right order). Now we are in our server :)
+
+## Update and Upgrade the system
```
Missing newline after header

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,214 @@
+```bash
+sudo apt update && sudo apt upgrade -y
+```
+We are using 'sudo' since I do not know if you're running as root user or as regular user. When you're logged in as regular user. No problem! Just enter your password when the terminal asks for your password. With the command above you will update the connected repository's the server is connected to and when the server is done updating the repository's it will directly upgrade all the packages w ...
````
Something here doesn't flow/make sense.
The guide should specify whether to work as `root` or regular user.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+# Installing the webserver
+
+Now we will get to the more difficult part of the entire server setup, installing nginx, php-fpm, mariadb-server with phpmyadmin.
```
Omit `Now we will get to the more difficult part of the entire server setup` - not informative.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,214 @@
+```bash
+sudo apt install nginx php-fpm mariadb-server -y
+```
+The installation process is now triggered.When this is finished we will install phpmyadmin to handle our database server
````
Typos

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+## Configuring the webserver
+
+First we want to know which version of php-fpm is installed we will check this by writing the following command in the terminal
```
Package names should be marked as "code" with the tilde, e.g.: `php-fpm`

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,214 @@
+    with Zend OPcache v7.2.24-0ubuntu0.18.04.3, Copyright (c) 1999-2018, by Zend Technologies
+```
+
+Second we want to do is enabling php in our nginx server and setting the configuration just as we want. So we will be going to edit our nginx config. Now we need to install a command line text editor called "vim". After that we installed it we will directly open the configuration file.
````
`PHP` not `php`

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,214 @@
+
+Second we want to do is enabling php in our nginx server and setting the configuration just as we want. So we will be going to edit our nginx config. Now we need to install a command line text editor called "vim". After that we installed it we will directly open the configuration file.
+```bash
+sudo apt install vim -y && sudo vim /etc/nginx/sites-available/default
````
No need to chain commands, just put them in two separate lines.
Explaining what `&&` does is unnecessary here, so let's just not use it.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
````diff
@@ -0,0 +1,214 @@
+sudo apt install vim -y && sudo vim /etc/nginx/sites-available/default
+```
+
+I am going to give you a few options in the configuration. Underneath here you will find the default configuration file of nginx. ( I only took the part of the config file we are in need to use and left out all the comments which we do use in this tutorial )
````
```suggestion
Below you will find the default configuration file of nginx (only relevant parts are shown, comments ommitted).
```

DSpeichert replied:
Try to avoid first-person narrative ("I did"), author is not important here.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+## Changing the web-root
+
+We can change the 'root' directory of the website folder, if you're only using your server as root leave it in '/var/www/html'. When using it as for example 'john' you can change the directory to your default directory when you come when you login which is: '/home/john' in the default directory of john you can do 'mkdir www' to create a directory for the website and set: '/home/john/www' on line  ...
```
On Linux, the term in use is `directory`, not `folder`. Since this is technical documentation, you have to try to use proper terms.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+I am going to give you a few options in the configuration. Underneath here you will find the default configuration file of nginx. ( I only took the part of the config file we are in need to use and left out all the comments which we do use in this tutorial )
+
+## Changing the web-root
```
Generally this section is not "necessary" as most people and guides do indeed keep the web root in `/var/www/` / `/var/www/html/` and rarely ever people need to put more than one "website" on a VPS for OTS.

DSpeichert commented on running-your-first-ot-server/setup-webserver-linux.md:
```diff
@@ -0,0 +1,214 @@
+
+Directly copy'd from the TFS github wiki(https://github.com/otland/forgottenserver/wiki/Compiling-on-Ubuntu):
+
+## Install the required software
```
The content from this line is duplicated - it's already documented in the TFS wiki, which we might move over to the OTS Guide.

It's not wrong but it will quickly get outdated. It's best to link to the wiki which is kept up to date.
In fact, the package list is already outdated below and the instructions wouldn't work.
